File: app/src/main/java/com/example/tetrisapp/interfaces/TetrisInterface.java
package com.example.tetrisapp.interfaces;

import com.example.tetrisapp.model.game.Piece;
import com.example.tetrisapp.model.game.Playfield;

import java.util.LinkedList;

public interface TetrisInterface {
    Piece getShadow();
    Piece getCurrentPiece();
    String getHeldPiece();
    PlayfieldInterface getPlayfield();
    PieceConfiguration getConfiguration();
    LinkedList<String> getTetrominoSequence();
    boolean isLockDelay();
    long getDelay();
    int getScore();
    int getLines();
    int getLevel();
    int getCombo();
}

File: app/src/main/java/com/example/tetrisapp/interfaces/TetrisInterface.kt
package com.example.tetrisapp.interfaces

import com.example.tetrisapp.model.game.Piece
import java.util.*

interface TetrisInterface {
    var shadow: Piece?
    var currentPiece: Piece?
    var heldPiece: String?
    var playfield: PlayfieldInterface
    var configuration: PieceConfiguration
    var tetrominoSequence: LinkedList<String>
    var isLocking: Boolean
    var score: Int
    var lines: Int
    var level: Int
    var combo: Int
    val delay: Long
}